LethalAPI.Core/Plugin.cs

```diff
@@ -10,32 +10,58 @@ namespace LethalAPI.Core;
 // ReSharper disable InconsistentNaming
 // ReSharper disable MemberCanBePrivate.Global
 #pragma warning disable SA1401 // field should be made private
+using System;
 
 using BepInEx;
 using BepInEx.Logging;
 using HarmonyLib;
+using MEC;
 
 /// <inheritdoc />
 [BepInPlugin(PluginInfo.PLUGIN_GUID, PluginInfo.PLUGIN_NAME, PluginInfo.PLUGIN_VERSION)]
 public class Plugin : BaseUnityPlugin
 {
+    /// <summary>
+    /// Gets the singleton for a plugin.
+    /// </summary>
+    public static Plugin Singleton = null!;
+
+    /// <summary>
+    /// Gets the <see cref="BepInEx.Logging.Logger"/>.
+    /// </summary>
     /// <summary>
     /// The base logger.
     /// </summary>
-    internal static ManualLogSource Log;
+#pragma warning disable SA1309
+    internal static ManualLogSource _Logger = null!;
+#pragma warning restore SA1309
 
     /// <summary>
     /// The harmony instance.
     /// </summary>
-    internal static Harmony Harmony;
+    internal static Harmony Harmony = null!;
 
     private void Awake()
     {
-        Log = Logger;
-
-        Log.LogInfo($"{PluginInfo.PLUGIN_GUID} is being loaded...");
+        _Logger = this.Logger;
         Harmony = new(PluginInfo.PLUGIN_GUID);
 
-        Harmony.PatchAll(typeof(Plugin).Assembly);
+        // Events..cctor -> Patcher.PatchAll will do the patching. This is necessary for dynamic patching.
+        _ = new Events.Events();
```

Comment:
Isn't it a decent idea to still save the Events instance?

Reply:
This happens as a singleton instance in the class. I find it easier to manage, as personally I like being able to do `Events.Events.Singleton.X` The class sets the `singleton = this` in the constructor.

Reply:
Fair enough.